### ISSUE TITLE
Consensus sequence with Ns

### ISSUE THREAD

**Issue by vmikk** [torognes/vsearch#181]
Dear Dr. Rognes and the VSEARCH team,

Probably there is a small bug with the consensus sequence computation for OTU clusters where the ambiguous nucleotide symbol (N) will be substituted to A.

Unfortunately we were not able to merge paired-end reads due to the insufficient length of overlap and we decided to work only with forward (R1) fragment. Because the length of the reads may vary due to quality trimming we decided to follow recommendations by USEARCH (more information is [here](http://drive5.com/usearch/manual/cmd_fastx_truncate.html) and [here](http://drive5.com/usearch/manual/global_trimming.html)) and to pad shorter sequences with Ns to avoid the terminal gaps problem.
After the clustering with VSEARCH (v1.11.1_linux_x86_64) everything is fine with centroid sequences. However all the N-tails will be converted to the A-tails in a consensus sequences.

Here is a small reproducible example:
```
cat > INPUT.fasta <<'EOT'
>seq1;size=12843;
CGCCCGTCGCTACTACCGATTGAATGGCTTAGTGAGGTCTCCGGATTGACTTTGGCGCACCGGCAACGGCATGCTATTGCTGAGAAGTTGATCAAACTTGGTCATTTAGAGGAAGTAAAAGTCGTAACAAGGTTTCC
>seq2;size=5236;
CGCCCGTCGCTACTACCGATTGAATGGCTTAGTGAGACCTCCGGATTGGCTTTGGGGAGTCGGAAACGACACCCTGTCGCTGAGAAGTTGGTCAAACTTGGTCATTTAGAGGAAGTAAAAGTCGTAACAAG
>seq3;size=4374;
CGCCCGTCGCTACTACCGATTGAATGGCTTAGTGAGGTCTCCGGATTGGCTTTGGGGAGCCGGCGACGGCACCCTATTGCTGAGAAGCTGATCAAACTTGGTCATTTAGAGGAAGTAAAAGTCGTAA
>seq4;size=3089;
CGCCCGTCGCTACTACCGATTGAATGGCTTAGTGAGGCTTTCGGATTGGATTTTGGCAGCTGGCAACAGCAGCTAGAAACTGAAAAGTTATCCAAACTTGGTCATTTAGAGGAAGTAAAAGTCC
EOT

~/usearch8.1.1861_i86linux32 \
  -fastx_truncate INPUT.fasta  \
  -trunclen 137 \
  -padlen 138 \
  -fastaout NNN.fasta

vsearch \
  --cluster_fast NNN.fasta \
  --id 0.97 \
  --centroids RES_Centroid.fasta \
  --consout RES_Consensus.fasta \
  --sizein --sizeout \
  --fasta_width 0

head RES_Consensus.fasta
```

What do you think about this N-padding approach? Or is it better just to trim all sequences to the common length and avoid Ns?

With best regards,
Vladimir

**Comment by frederic-mahe** [comment]
Hi Vladimir,

thanks a lot for the reproducible example and the detailed explanations. Just to be sure, the expected behavior is to preserve the N-tails in the consensus, correct?

**Comment by vmikk** [comment]
@frederic-mahe 
Thank you for the reply. Yes, we expected that the N-tail will be preserved in the consensus.

**Comment by torognes** [comment]
Hi! Thanks for reporting this problem. I think we have to consider this as a bug. I will fix it in the next release.

**Comment by torognes** [comment]
This issue is fixed in version 1.11.2 just released.

**Comment by vmikk** [comment]
@torognes Thank you very much!
